File: 8_kyu/To_squareroot_or_not_to_squareroot.jl
"""
  Returns an array with the root of each item if the item is a perfect square, otherwise the item squared.
"""
function square_or_square_root(arr::Array{T})::Array{T} where T<:Real
  function s_or_r(x::T)::T
    r = sqrt(x)
    isinteger(r) ? T(r) : x^2
  end
  map(s_or_r, arr)
end

File: 8_kyu/To_squareroot_or_not_to_squareroot.py
def square_or_square_root(arr: list[int]) -> list[int]:
    """
    Returns list with each item square or square root.
    """
    return [s_or_s(x) for x in arr]

def s_or_s(x :int) -> int:
    """
    Returns square root of x if it is an integer, else return x squared.
    """
    r = x**0.5
    return int(r) if r%1 == 0 else x**2